modules/programs/aerospace.nix

```diff
@@ -0,0 +1,231 @@
+                app-name-regex-substring = "CoolApp";
+                during-aerospace-startup = false;
+              };
+              check-further-callbacks = false;
+              run = [ "move-node-to-workspace m" "resize-node" ];
+            }];
+            description =
+              "Commands to run every time a new window is detected with optional conditions.";
+          };
+          workspace-to-monitor-force-assignment = mkOption {
+            type = with types; attrsOf (oneOf [ int str (listOf str) ]);
+            default = { };
+            description = ''
+              Map workspaces to specific monitors.
+              Left-hand side is the workspace name, and right-hand side is the monitor pattern.
+            '';
+            example = {
+              "1" = 1; # First monitor from left to right.
+              "2" = "main"; # Main monitor.
+              "3" = "secondary"; # Secondary monitor (non-main).
+              "4" = "built-in"; # Built-in display.
+              "5" =
+                "^built-in retina display$"; # Regex for the built-in retina display.
+              "6" = [ "secondary" "dell" ]; # Match first pattern in the list.
+            };
+          };
+          on-focus-changed = mkOption {
+            type = with types; listOf str;
+            default = [ ];
+            example = [ "move-mouse monitor-lazy-center" ];
+            description =
+              "Commands to run every time focused window or workspace changes.";
+          };
+          on-focused-monitor-changed = mkOption {
+            type = with types; listOf str;
+            default = [ "move-mouse monitor-lazy-center" ];
+            description = "Commands to run every time focused monitor changes.";
+          };
+          exec-on-workspace-change = mkOption {
+            type = with types; listOf str;
+            default = [ ];
+            example = [
+              "/bin/bash"
+              "-c"
+              "sketchybar --trigger aerospace_workspace_change FOCUSED=$AEROSPACE_FOCUSED_WORKSPACE"
+            ];
+            description = "Commands to run every time workspace changes.";
+          };
+          key-mapping.preset = mkOption {
+            type = types.enum [ "qwerty" "dvorak" ];
+            default = "qwerty";
+            description = "Keymapping preset.";
+          };
+        };
+      };
+      default = { };
+      example = lib.literalExpression ''
+        {
+          gaps = {
+            outer.left = 8;
+            outer.bottom = 8;
+            outer.top = 8;
+            outer.right = 8;
+          };
+          mode.main.binding = {
+            alt-h = "focus left";
+            alt-j = "focus down";
+            alt-k = "focus up";
+            alt-l = "focus right";
+          };
+        }
+      '';
+      description = ''
+        AeroSpace configuration, see
+        <link xlink:href="https://nikitabobko.github.io/AeroSpace/guide#configuring-aerospace"/>
```

Comment:
Should be markdown: `<https://nikitabobko.github.io/AeroSpace/guide#configuring-aerospace>`